temporian/implementation/numpy/data/event_set.py

```diff
@@ -57,9 +64,161 @@
 }
 
 
+def is_supported_numpy_dtype(numpy_dtype) -> bool:
```

Comment:
I would make the API opt-in. I,e, we define somewhere the small set of class / functions that are part of the API.
This way, by default, a function defined in temporian/implementation/numpy/data/event_set.py should not be part of the API.

Reply:
Taking this to tomorrow's meeting to discuss, but I would move it to the bottom for now so that the docs make more sense. Also, IMO as a developer it's best to be able to easily find the EventSet class at the top of its file, rather than the aux functions?